src/BCReader.jl

```diff
@@ -182,22 +182,54 @@ The times for which data is extracted depends on the specifications in the
 function update_midmonth_data!(date, bcf_info::BCFileInfo{FT}) where {FT}
     # monthly count
     (; bcfile_dir, comms_ctx, hd_outfile_root, varname, all_dates, scaling_function) = bcf_info
-    midmonth_idx = bcf_info.segment_idx[1]
-    midmonth_idx0 = bcf_info.segment_idx0[1]
-
-    if (midmonth_idx == midmonth_idx0) && (Dates.days(date - all_dates[midmonth_idx]) < 0) # for init
-        midmonth_idx = bcf_info.segment_idx[1] -= Int(1)
-        midmonth_idx = midmonth_idx < Int(1) ? midmonth_idx + Int(1) : midmonth_idx
-        @warn "this time period is before BC data - using file from $(all_dates[midmonth_idx0])"
-        bcf_info.monthly_fields[1] .= scaling_function(
-            Regridder.read_from_hdf5(bcfile_dir, hd_outfile_root, all_dates[Int(midmonth_idx0)], varname, comms_ctx),
-            bcf_info,
-        )
-        bcf_info.monthly_fields[2] .= deepcopy(bcf_info.monthly_fields[1])
-        bcf_info.segment_length .= Int(0)
-
-    elseif Dates.days(date - all_dates[end - 1]) > 0 # for fini
-        @warn "this time period is after BC data - using file from $(all_dates[end - 1])"
+    segment_idx = bcf_info.segment_idx[1] # index of the current date in the file. [segment_idx, segment_idx+1] indexes the current segment between which we interpolate
+    segment_idx0 = bcf_info.segment_idx0[1] # reference index (first segment_idx - 1)
+
+    # upon initialization (segment_idx == segment_idx0) and not initialized after final file date
+    if (segment_idx == segment_idx0) && !((date - all_dates[end]).value >= 0)
+        #  case 1: model date is before the first segment read from file
+        if (date - all_dates[segment_idx]).value < 0
```

Comment:
should this be `segment_idx0` if we want it to be the first segment read from the file?